src/compiler/transformers/jsx.ts

```diff
@@ -114,7 +114,7 @@ namespace ts {
                 compilerOptions.reactNamespace,
                 tagName,
                 objectProperties,
-                filter(map(children, transformJsxChildToExpression), isDefined),
+                children && mapDefined(children, transformJsxChildToExpression),

```

Comment:
Pretty much all of our other array utils in _core.ts_ test for the array possibly being undefined. Should we not also do that for `mapDefined`?